### PR TITLE
Better CLI Arguments parsing

### PR DESCRIPTION
**Pull request by bararchy**
Using OptionParser

### CODE REVIEW

File: src/cryload.cr [hunk 1]
```diff
@@ -38,16 +39,28 @@ module Cryload
...
+OptionParser.parse(ARGV) do |opts|
+  opts.banner = "Usage: ./cryload [options]"
+  
+  opts.on("-s SERVER", "--server SERVER", "Target Server") do |v|
+    options[:server] = v
   end
-  puts "Preparing to make it CRY for #{request_count} requests!".colorize(:green)
-  Cryload::LoadGenerator.new host, request_count.to_i
+
+  opts.on("-r REQUESTS", "--requests REQUESTS", "Number of requests to make") do |v|
```
Comment (sdogruyol) [inline src/cryload.cr]: How about changing this to -n numbers? Similar to `ab`

**Comment by sdogruyol** [comment]
Also we need to set a default request number otherwise if the user does something like

`./cryload -h http://www.example.com`

There will be no output and no error. Let's make the default 1000 :+1:

**Comment by bararchy** [comment]
@Sdogruyol Done :)

**Comment by bararchy** [comment]
@Sdogruyol Do you want me to enforce inputing -s ? (as in server)

File: src/cryload.cr [hunk 2]
```diff
@@ -38,16 +39,29 @@ module Cryload
...
+  opts.banner = "Usage: ./cryload [options]"
+  
+  opts.on("-s SERVER", "--server SERVER", "Target Server") do |v|
+    options[:server] = v
   end
-  puts "Preparing to make it CRY for #{request_count} requests!".colorize(:green)
-  Cryload::LoadGenerator.new host, request_count.to_i
+
+  opts.on("-n NUMBERS", "--numbers NUMBERS", "Number of requests to make") do |v|
+    options[:requests] = v
```
Comment (sdogruyol) [inline src/cryload.cr]: Let's change this to :numbers also :+1:

**Comment by sdogruyol** [comment]
On empty command you already tell the options to run so that'd be not necessary. But if you could enforce -n that'd be awesome :+1: 

Can you also squash the commits :hammer:

**Comment by bararchy** [comment]
@Sdogruyol Thats the best I can do with squashing XD

**Comment by sdogruyol** [comment]
Oops now that's 7 commit how did you squash :smile:

**Comment by sdogruyol** [comment]
Not specifying numbers is giving the wrong output

```
./cryload -s http://localhost:8080
You have to specify '-n' and '-s' flags, for help use '-h'
```

**Comment by bararchy** [comment]
@Sdogruyol Fixed.
How do I sqush commits ? I used `git rebase -i HEAD~4`

**Comment by sdogruyol** [comment]
@bararchy http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by bararchy** [comment]
@Sdogruyol ok.... it seems I just cant do it :( maybe you can just take the changes and applied them by hand ?

**Comment by sdogruyol** [comment]
Wow this branch is messed up now :dancer: 

Can you open a new pr in a new branch with your final changes it'll be much more clearer in the git history :+1:

**Pull request closed without merging**